taichi/ir/ir_builder.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "taichi/ir/ir.h"
+
+TLANG_NAMESPACE_BEGIN
+
+class IRBuilder {
+ private:
+  struct InsertPoint {
+    Block *block;
+    int position;
+  };
+
+  std::unique_ptr<IRNode> root_;
+  InsertPoint insert_point_;
+
+ public:
+  IRBuilder();
+
+  // General inserter. Returns stmt.get().
+  Stmt *insert(std::unique_ptr<Stmt> &&stmt);
+
+  // Constants. TODO: add more types
+  Stmt *get_int32(int32 value);
```

Comment:
You may also consider a `template <typename T> Stmt *get_const(T val)` template function and then specialize for different types (or use `if constexpr (std::is_same_v<T, int32>)`). Just a random idea and the decision is yours :-)